src/Pusher.php

```diff
@@ -1073,4 +1006,153 @@ public function ensure_valid_signature(array $headers, string $body): void
     {
         $this->verifySignature($headers, $body);
     }
+
+    /**
+     * Returns an event represented by an associative array to be used in creating events and batch_events requests
+     *
+     * @param array|string $channels A channel name or an array of channel names to publish the event on.
+     * @param string $event
+     * @param mixed $data Event data
+     * @param array $params [optional]
+     * @param bool $already_encoded [optional]
+     *
+     * @throws PusherException
+     *
+     * @return array Event associative array
+     */
+    private function make_event(array $channels, string $event, $data, array $params = [], ?string $info = null, bool $already_encoded = false): array
+    {
+      $has_encrypted_channel = false;
+        foreach ($channels as $chan) {
+            if (PusherCrypto::is_encrypted_channel($chan)) {
+                $has_encrypted_channel = true;
+                break;
+            }
+        }
+
+        if ($has_encrypted_channel) {
+            if (count($channels) > 1) {
+                // For rationale, see limitations of end-to-end encryption in the README
+                throw new PusherException('You cannot trigger to multiple channels when using encrypted channels');
+            } else {
+                try {
+                    $data_encoded = $this->crypto->encrypt_payload(
+                        $channels[0],
+                        $already_encoded ? $data : json_encode($data, JSON_THROW_ON_ERROR)
+                    );
+                } catch (\JsonException $e) {
+                    throw new PusherException('Data encoding error.');
+                }
+            }
+        } else {
+            try {
+                $data_encoded = $already_encoded ? $data : json_encode($data, JSON_THROW_ON_ERROR);
+            } catch (\JsonException $e) {
+                throw new PusherException('Data encoding error.');
+            }
+        }
+
+        // json_encode might return false on failure
+        if (!$data_encoded) {
+            $this->log('Failed to perform json_encode on the the provided data: {error}', [
+                'error' => print_r($data, true),
+            ], LogLevel::ERROR);
+        }
+
+        $post_params = [];
+        $post_params['name'] = $event;
+        $post_params['data'] = $data_encoded;
+        $channel_values = array_values($channels);
+        if (count($channel_values) == 1) {
+          $post_params['channel'] = $channel_values[0];
+        } else {
+          $post_params['channels'] = $channel_values;
+        }
```

Comment:
This is a backward breaking change. The payload has changed from being `channel` in case only one channel is supplied. This has always been `channels` no matter the number of channels provided.

Ref: https://github.com/pusher/pusher-http-php/issues/351

Reply:
I'm missing something. Can you help me understand how this breaks things for you? This payload is sent to the pusher server and it accepts both.

Reply:
You are right, this only applies to custom servers.